spec/support/request/shop_workflow.rb

```diff
@@ -34,6 +49,58 @@ def add_product_to_cart(order, product, quantity: 1)
     order.update_distribution_charge!
   end
 
+  # Add an item to the cart
+  #
+  # At the moment, the user enters the quantity into an input field.
+  # But with the coming mobile-friendly UX, the user will click a button to
+  # add an item, hence the naming.
+  #
+  # This is temporary code. The duplication will be removed by the mobile
+  # product listings feature. This has been backported to avoid merge
+  # conflicts and to make the current build more stable.
```

Comment:
This type of comment should never land on the codebase. The fact that is temporary doesn't need to be there. I'd remove this comment from the code and add it as a comment on the PR.

Reply:
My reason for the comment was so that if someone sees the code later that they don't start refactoring it. Somebody may come across the code without seeing the commit. I'll leave it for now. It will be removed soon anyway.